### PR TITLE
Add is-active class to nav bar element when related ui-sref is active.

### PR DESCRIPTION
**Pull request by jonman**
TP: https://rockabox.tpondemand.com/entity/9146

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 ### Added
 
 - [`rb-datetime-control` show date only](https://github.com/rockabox/rbx_ui_components/pull/205)
+- [`rb-nav-bar` is-active states](https://github.com/rockabox/rbx_ui_components/pull/210) Add is-active class when the related ui-sref directive's state is active and remove when it is inactive.
```
Comment (chrisvxd) [inline CHANGELOG.md]: any thoughts on a more explicit name for this change? something like `rb-nav-bar` sets `is-active` based on state

File: test/unit/rb-nav-bar/rb-nav-bar.spec.js [hunk 2]
```diff
@@ -3,77 +3,89 @@ define([
...
             it('should set the target if set', function () {
-                var navBar = angular.element(template),
-                    element = $compile(navBar)($scope),
-                    options;
+                compileTemplate(template);
 
-                $scope.$apply();
-                options = angular.element(element.find('a'));
+                var options = angular.element(element.find('a'));
                 expect(angular.element(options[2]).attr('target')).toBe('_self');
```
Comment (chrisvxd) [inline test/unit/rb-nav-bar/rb-nav-bar.spec.js]: Mind keeping these test refactors in a separate commit? Not fussed about PR but would like to keep a clean commit history.

File: src/rb-nav-bar/rb-nav-bar-directive.js [hunk 3]
```diff
@@ -17,11 +17,6 @@ define([
      *           Option objects should be defined as follows:
      *           {id: '<an identifier for the option>',
      *           text: 'Text to be displayed'}
-     *  'clickfunction': a custom function to be executed on every option when clicked
-     *                   The function receives the 'id' from the option object as param
-     *  'isactivefunction': a custom function that returns if an option should be displayed
-     *                   with an `is-active` class.
-     *                   The function receives the 'id' from the option object as param
```
Comment (chrisvxd) [inline src/rb-nav-bar/rb-nav-bar-directive.js]: Same as comment below

File: test/unit/rb-nav-bar/rb-nav-bar.spec.js (hunk 2 above)
Comment (jonman) [inline test/unit/rb-nav-bar/rb-nav-bar.spec.js]: Tests will fail for that commit if I moved them to different commits.

File: CHANGELOG.md (hunk 1 above)
Comment (jonman) [inline CHANGELOG.md]: Will update

File: test/unit/rb-nav-bar/rb-nav-bar.spec.js (hunk 2 above)
Comment (chrisvxd) [inline test/unit/rb-nav-bar/rb-nav-bar.spec.js]: @jonman why would they fail? the tests themselves haven't changes besides use of helpers

Comment (jonman) [inline test/unit/rb-nav-bar/rb-nav-bar.spec.js]: `$state` dependency within the directive.

File: src/rb-nav-bar/rb-nav-bar-directive.js (hunk 3 above)
Comment (jonman) [inline src/rb-nav-bar/rb-nav-bar-directive.js]: Updated :+1:
